src/null-safety/index.md

```diff
@@ -5,20 +5,45 @@ description: Information about Dart's null safety feature
 description: Dart 空安全的有关内容
 ---
 
-The Dart language now supports sound null safety!
+The Dart language comes with sound null safety.
+
+Null safety prevents errors that result from unintentional access
+of variables set to `null`.
+For example, if a method expects an integer but receives `null`,
+your app causes a runtime error. This type of error, a null dereference error,
+can be difficult to debug.
+
+With sound null safety variables are 'non-nullable' by default:
+They can be assigned only values of the declared type
+(e.g. `int i=42`), and never be assigned `null`.
+You can specify that a type of a variable is nullable
+(e.g. `int? i`),
+and only then can they contain either a `null` *or*
+a value of the defined type.
+
+Sound null safety changes potential **runtime errors**
+into **edit-time** analysis errors, by flagging when
+any non-nullable variable hasn't been initialized with a 
+non-null value or is being assigned a `null`.
+This allows you to fix these errors before deploying your app.
+
+{{site.alert.warn}}
+In Dart 2.x SDKs, you can enable or disable sound null safety 
+through configuration of the project SDK constraint.
+To learn more, see [Enabling/disabling null safety](#enable-null-safety).
+
+Dart 3--planned for a mid-2023 release--
+will require sound null safety. It will prevent code from running without it.
+All existing code must be [migrated](#migrate) to sound null safety
+to be compatible with Dart 3.
+To learn more, see the [Dart 3 sound null safety tracking issue][].
+{{site.alert.end}}
```

Comment:
```suggestion
{{site.alert.warn}}

  In Dart 2.x SDKs, you can enable or disable sound null safety 
  through configuration of the project SDK constraint.
  To learn more, see [Enabling/disabling null safety](#enable-null-safety).

  在 Dart 2.x 版本的 SDK 里，你可以通过设置项目的 SDK 限制来控制是否启用空安全。
  你可以通过 [启用/禁用空安全](#enable-null-safety) 小节来了解更多。

  Dart 3--planned for a mid-2023 release--
  will require sound null safety. It will prevent code from running without it.
  All existing code must be [migrated](#migrate) to sound null safety
  to be compatible with Dart 3.
  To learn more, see the [Dart 3 sound null safety tracking issue][].

  预计在 2023 年中发布的 Dart 3 计划强制使用空安全。
  该版本不允许未使用空安全的代码运行。
  所有的代码都需要 [迁移](#migrate) 至健全的空安全，以适配 Dart 3。
  你可以通过 [Dart 3 空安全问题跟踪][Dart 3 sound null safety tracking issue]
  来了解更多。

{{site.alert.end}}
```